jni/src/common/Foreign/JNI/Unsafe.hs

```diff
@@ -472,6 +478,8 @@ newJVM options = JVM_ <$> do
 -- | Deallocate a 'JVM' created using 'newJVM'.
 destroyJVM :: JVM -> IO ()
 destroyJVM (JVM_ jvm) = do
+    readIORef globalRefCleaner >>= stopGlobalRefCleaner
+    writeIORef globalRefCleaner (error "Cannot delete a reference: a dedicated thread is not initialized")
```

Comment:
Maybe we should have
```haskell
uninitializedGlobalRefCleaner :: GlobalRefCleaner
uninitializedGlobalRefCleaner = error "Cannot delete a reference: a dedicated thread is not initialized"
```
since it would be used twice.